graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +338,182 @@ def start_global_operations(self):
 
     def filter(self, predicate):
         """Execute a Filter Block."""
-        sql_expression = predicate.to_sql(self._aliases, self._current_alias)
+        left_predicate_folded = isinstance(predicate.left, FoldedContextField)
+        right_predicate_folded = isinstance(predicate.right, FoldedContextField)
+        if self._in_fold or left_predicate_folded or right_predicate_folded:
+            raise NotImplementedError('Filters inside a fold are not implemented yet.')
+
+        sql_expression = predicate.to_sql(self._aliases,
+                                          self._current_alias)
         if self._is_in_optional_scope():
             sql_expression = sqlalchemy.or_(sql_expression,
                                             self._came_from[self._current_alias].is_(None))
         self._filters.append(sql_expression)
 
+    def get_join_descriptor(self, fold_scope_location):
```

Comment:
I'd put the word fold in the function name so it is clear that this function is only for folds since the current name makes it sound pretty general.